ApiFirst/ApiFirst/Controllers/CostumersController.cs

```diff
@@ -0,0 +1,90 @@
+﻿using DomainModels.Models;
+using DomainServices.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CustomersController : ControllerBase
+    {
+        private readonly ICustomerService _customerService;
+
+        public CustomersController(ICustomerService customerService)
+        {
+            _customerService = customerService ?? throw new ArgumentNullException(nameof(customerService));
+        }
+
+        [HttpGet("cpf/{cpf}")]
+        public Customer? GetByCpf(string cpf)
+        {
+            return _customerService.GetByCpf(cpf);
+        }
+
+
+        [HttpGet]
+        public List<Customer> GetAll()
+        {
+            return _customerService.GetAll();
+        }
+
+        [HttpGet("{id}")]
+        public ActionResult<Customer?> GetById(int id)
+        {
+
+            var response = _customerService.GetById(id);
+
+            return response is null
+                ? NotFound($"Não foi encontrado Customer para o Id: {id}")
+                : Ok(response);
+        }
+
+        [HttpPost]
+        public IActionResult Create([FromBody] Customer model)
+        {
+            return _customerService.Create(model)
+                ? CreatedAtAction(nameof(GetById), new { id = model.Id }, model)
```

Comment:
```suggestion
                ? Created("", model.Id)
```
Aqui devemos retornar apenas o ID do novo Customer criado